src/oss-diff/DiffTool.cs

```diff
@@ -0,0 +1,207 @@
+                    int afterCount = 0;
+                    foreach (var line in diff.Lines)
+                    {
+                        switch (line.Type)
+                        {
+                            case ChangeType.Inserted:
+                                if (!options.RemovedOnly || options.AddedOnly)
+                                {
+                                    if (beforeBuffer.Any())
+                                    {
+                                        foreach (var buffered in beforeBuffer)
+                                        {
+                                            Console.WriteLine($"  {buffered.Pastel(Color.Gray)}");
+                                        }
+                                        beforeBuffer.Clear();
+                                    }
+                                    afterCount = options.After;
+                                    Console.WriteLine($"+ {line.Text}".Pastel(Color.Green));
+                                }
+                                break;
+                            case ChangeType.Deleted:
+                                if (!options.AddedOnly || options.RemovedOnly)
+                                {
+                                    if (beforeBuffer.Any())
+                                    {
+                                        foreach (var buffered in beforeBuffer)
+                                        {
+                                            Console.WriteLine($"  {buffered.Pastel(Color.Gray)}");
+                                        }
+                                        beforeBuffer.Clear();
+                                    }
+                                    afterCount = options.After;
+                                    Console.WriteLine($"- {line.Text}".Pastel(Color.Red));
+                                }
+                                break;
+                            default:
+                                if (options.Context == -1)
+                                {
+                                    Console.WriteLine($"  {line.Text.Pastel(Color.Gray)}");
+                                }
+                                else if (afterCount-- > 0)
+                                {
+                                    Console.WriteLine($"  {line.Text.Pastel(Color.Gray)}");
+                                }
+                                else
+                                {
+                                    beforeBuffer.Add(line.Text);
+                                    while(beforeBuffer.Count > options.Before)
+                                    {
+                                        beforeBuffer.RemoveAt(0);
+                                    }
+                                }
+
+                                break;
+                        }
+                    }
+                }
+
+                if (options.DeleteAfterDiff)
```

Comment:
Is this flag kind of the same as caching? Allowing caching means the packages wont be deleted after download, in addition to not downloading if present in the first place - that's the behavior in other tools, might make to just keep it consistent.

Reply:
Fair point. I'll use the cache argument instead.